lib/datadog/statsd/connection.rb

```diff
@@ -12,6 +12,7 @@ def reset_telemetry
         telemetry.reset
       end
 
+      # not thread safe: callers from multiple threads must synchronize calls to this method
       def write(payload)
         logger.debug { "Statsd: #{payload}" } if logger
 
```

Comment:
⬇️  GitHub doesn't let me comment on line 34, but it appears to me that if we always try to connect before `send_message`, then the `Connection` class no longer even needs to know about `#connect`: it effectively becomes a private implementation detail of the subclasses.

The remaining contract we have is that we can call `close` and then expect a subsequent `send_message` to still work.

Reply:
You are right, but, I like the clarity of https://github.com/DataDog/dogstatsd-ruby/pull/214/files#diff-01cb409d6d5a1fb2aebe52e78e877a91e3f6654758931899427a43656211a073R32-R35
You can explicitly read in the retry mechanism that it is closing the connection, re-opening it, and retrying.
I know the `connect` isn't needed since `send_message` will do it.
Not a strong opinion though, I can remove the `connect` here.
WDYT?